src/vision.cs

```diff
@@ -915,6 +1247,32 @@ interface VNRequest : NSCopying {
 
 		[Export ("usesCPUOnly")]
 		bool UsesCpuOnly { get; set; }
+
+		[TV (12,0), Mac (10,14, onlyOn64: true), iOS (12,0)]
+		[Export ("revision")]
+		VNRequestRevision Revision { get; set; }
+
+		// We must inline the following 4 static properties ('WeakSupportedRevisions', 'SupportedRevisions', 'DefaultRevision' and 'CurrentRevision')
+		// into subclasses so the correct class_ptr is used.
+		[TV (12,0), Mac (10,14, onlyOn64: true), iOS (12,0)]
+		[Static]
+		[Export ("supportedRevisions", ArgumentSemantic.Copy)]
+		NSIndexSet WeakSupportedRevisions { get; }
+
+		[TV (12,0), Mac (10,14, onlyOn64: true), iOS (12,0)]
+		[Static]
+		[Wrap ("GetSupportedVersions (WeakSupportedRevisions)")]
+		VNRequestRevision [] SupportedRevisions { get; }
+
+		[TV (12,0), Mac (10,14, onlyOn64: true), iOS (12,0)]
+		[Static]
+		[Export ("defaultRevision")]
+		VNRequestRevision DefaultRevision { get; }
+
+		[TV (12,0), Mac (10,14, onlyOn64: true), iOS (12,0)]
+		[Static]
+		[Export ("currentRevision")]
+		VNRequestRevision CurrentRevision { get; }
```

Comment:
So `VNRequest` is `abstract` and I do not see how calling those static methods (on it) would help - since they are redefined in the concrete subclasses.

What about removing them from this base class (unusable) and only exposing them in the subclass (no need for `[New]`) with a different signature that use an enum that's specific for each of them ?

Reply:
I haven't thought it through fully, but this tact sounds worth exploring. I'm not allowed to complain about Appkit after seeing this PR @dalexsoto  :)